### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by jrean**

### CODE REVIEW

File: docs/Installation.md [hunk 1]
```diff
@@ -1,68 +1,66 @@
 # Installation
-Follow this guide if you want to install this package in new\existing laravel installation.
+Follow this guide to install this package in a new or existing laravel project.
 
-Follow next steps one after another:
 1. [Laravel](#laravel)
 2. [Doctrine Setup](#doctrine-setup)
-3. [Package config](#package-config)
-4. [Verify installation](#verify-installation)
+3. [Package Installation](#package-install)
```
Comment (R3VoLuT1OneR) [inline docs/Installation.md]: The link must be "#package-installation" to match the ancor generated for the "Package Install" section.

File: docs/Installation.md [hunk 2]
```diff
@@ -1,68 +1,66 @@
...
+4. [Package Config](#package-config)
+5. [Verify installation](#verify-installation)
```
Comment (R3VoLuT1OneR) [inline docs/Installation.md]: Please remove this line as we don't have this section anymore.
I think we can skip the section at all.

File: docs/Installation.md (hunk 1 above)
Comment (jrean) [reply docs/Installation.md]: Fixed

File: docs/Installation.md (hunk 2 above)
Comment (jrean) [reply docs/Installation.md]: Fixed

**Comment by jrean** [comment]
@R3VoLuT1OneR Still WIP, do not merge yet. I'll let you know when it's ready to be merged.

File: docs/README.md [hunk 3]
```diff
@@ -1,33 +1,56 @@
...
+- Filtering
+- Create resources
+- Update resources
+- Update relationships
+- Delete resources
 
-### Setup
-Please follow the [instructions](./Installation.md) and set up the package in your Laravel installation.
+## Installation & Config
+1. [Laravel](./Installation.md#laravel)
```
Comment (R3VoLuT1OneR) [inline docs/README.md]: The link must be `./Installation.md#laravel-installation`

File: docs/README.md [hunk 4]
```diff
@@ -1,33 +1,56 @@
...
+2. [Doctrine Setup](./Installation.md#doctrine-setup)
```
Comment (R3VoLuT1OneR) [inline docs/README.md]: The link must be to `./LaravelDoctrine.md`

File: docs/ResourceInterface.md [hunk 5]
```diff
@@ -24,105 +26,76 @@ public function getId(): int
...
 
-```php
-->innerJoin("user.$mappedBy", 'relation')
-->where("relation.users = ${resource->getId()}")
-```
+- `property` (optional):
 
-#### property
-Optional parameter that is defining the entity property name, by default the `name` param value will be used.
+Override the default value of `{name}` for custom endpoints.
```
Comment (R3VoLuT1OneR) [inline docs/ResourceInterface.md]: It must be improved

It's can be something like:

"Override the relationship property name that will be the `{name}` by default."
for example if you have `protected Country $userCountry` but you want to have simple "country" relationship name, you need to provide "userCountry" as a value.

File: docs/ResourceInterface.md [hunk 6]
```diff
@@ -131,14 +104,30 @@ public static function relationships(): RelationshipsCollection
...
 
-The Trait provides static method `memoizeRelationships(callback $cb): RelationshipsCollection`.
-Method must receive callback that will return relationships' collection.
+The entity (resource) class being the relation.
+
+- `mappedBy` (required):
+
+Name of the association-field on the owning side of the relation.
+
+- `property` (optional):
```
Comment (R3VoLuT1OneR) [inline docs/ResourceInterface.md:151]: It must be improved

It's can be something like:

"Override the relationship property name that will be the {name} by default."
for example if you have `protected Country $userRoles` but you want to have simple "country" relationship name, you need to provide "userRoles" as a property param value.

File: docs/README.md (hunk 4 above)
Comment (jrean) [reply docs/README.md]: It was right but the pages hierarchy was confusing. I improved it.

File: docs/ResourceInterface.md (hunk 5 above)
Comment (jrean) [reply docs/ResourceInterface.md]: Great feedback. Thxs

File: docs/ResourceInterface.md [hunk 7]
```diff
@@ -24,121 +26,158 @@ public function getId(): int
...
 
-### To-One
-To define new "To-One" relationship we must add new [Sowl\JsonApi\Relationships\ToOneRelationship](../src/Relationships/ToOneRelationship.php).
+The entity (resource) class being the relation.
 
-To-One relationship returns single resource in the responses.
+- `property` (optional):
 
-2 required parameters must be provider and 1 optional.
+Allow to set a custom value for the `{name}` for custom endpoints.
```
Comment (R3VoLuT1OneR) [inline docs/ResourceInterface.md]: It's not working like that.

`{name}` - is the endpoint name and a name in the `data.relationships.{name} in the responses.
`{property}` - Default: {name} is the Entity property name that will be updated. 

The property is not for creating custom endpoints, it's for allowing custom {name} if the {name} not matching the entity property.

File: docs/ResourceInterface.md [hunk 8]
```diff
@@ -24,121 +26,158 @@ public function getId(): int
...
+    return new RelationshipsCollection([
+        ToOneRelationship::create('userCountry', Country::class)
+    ]);
+}
+```
 
-#### name
-First parameter must be the name of relationships that will be used for generating building endpoints.
-For example `GET /user/1/{name}` or `GET /user/1/relationships/{name}`
+Set the 3rd optional parameter `property` to `country` to have a custom endpoint `GET /user/1/country`.
```
Comment (R3VoLuT1OneR) [inline docs/ResourceInterface.md]: To get the custom endpoint `/user/1/country` then you need to set next:
```php
        ToOneRelationship::create('country', Country::class, 'userCountry')
```

So name will be "country" and the property "userCountry" as we have the `protected Country $userCountry`

**Pull request merged**